[18.0][IMP] account_statement_sheet_file: replace xlrd for openpyxl

andyengit opened the pull request:
Added openpyxl as a dependency for XLSX file handling and updated the parser to utilize openpyxl instead of xlrd.

https://github.com/pandas-dev/pandas/issues/28547

ecino committed:
[MIG] account_statement_import_online_wise: Migration to 14.0

ayushin committed:
[IMP] account_statement_import_camt: support datetime (DtTm) as booking date

Certain institutions (wise.com) use DtTm instead of Dt for transaction date,
this fix supports either original Dt or DtTm

Commit:
[UPD] Update account_statement_import_online_wise.pot

OCA-git-bot committed:
[BOT] post-merge updates

remytms committed:
[FIX] account_statement_import_base: sanitized acc number

The field acc_number in the database is not sanitized by default.

This leads to account_bank_statement_line with an empty partner_id
because the check done in _statement_line_import_update_hook()
uses the sanitized_acc_number instead of the acc_number.

For a right comparison the sanitized version of the acc_number should be
put in the speeddict.

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 5.5% (1 of 18 strings)

Translation: bank-statement-import-14.0/bank-statement-import-14.0-account_statement_import_online_wise
Translate-URL: https://translation.odoo-community.org/projects/bank-statement-import-14-0/bank-statement-import-14-0-account_statement_import_online_wise/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 11.1% (2 of 18 strings)

Translation: bank-statement-import-14.0/bank-statement-import-14.0-account_statement_import_online_wise
Translate-URL: https://translation.odoo-community.org/projects/bank-statement-import-14-0/bank-statement-import-14-0-account_statement_import_online_wise/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 16.6% (3 of 18 strings)

Translation: bank-statement-import-14.0/bank-statement-import-14.0-account_statement_import_online_wise
Translate-URL: https://translation.odoo-community.org/projects/bank-statement-import-14-0/bank-statement-import-14-0-account_statement_import_online_wise/it/

jbaudoux committed:
[FIX] account_statement_import_sheet_file: fix none separator

TDu committed:
Add account_statement_line_order

thomaspaulb committed:
[FIX] account_statement_import_online_wise: use foreign_currency_id

Foreign currency transactions failed to load because of the following check in Odoo:

https://github.com/OCA/OCB/blob/1bbabffeead7a00defe29942a5c9ecf574393dd6/addons/account/models/account_bank_statement.py#L851

Which says that when amount_currency is filled, foreign_currency_id should be filled too.

OCA-git-bot committed:
[BOT] post-merge updates

andyengit committed:
[18.0][IMP] account_statement_sheet_file: replace xlrd for openpyxl

Added openpyxl as a dependency for XLSX file handling and updated the parser to utilize openpyxl instead of xlrd.

https://github.com/pandas-dev/pandas/issues/28547

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (160 of 160 strings)

Translation: bank-statement-import-18.0/bank-statement-import-18.0-account_statement_import_online_paypal
Translate-URL: https://translation.odoo-community.org/projects/bank-statement-import-18-0/bank-statement-import-18-0-account_statement_import_online_paypal/it/

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (16 of 16 strings)

Translation: bank-statement-import-18.0/bank-statement-import-18.0-account_statement_import_online_plaid
Translate-URL: https://translation.odoo-community.org/projects/bank-statement-import-18-0/bank-statement-import-18-0-account_statement_import_online_plaid/it/

bo-eska committed:
Translated using Weblate (Turkish)

Currently translated at 97.1% (101 of 104 strings)

Translation: bank-statement-import-18.0/bank-statement-import-18.0-account_statement_import_sheet_file
Translate-URL: https://translation.odoo-community.org/projects/bank-statement-import-18-0/bank-statement-import-18-0-account_statement_import_sheet_file/tr/

bo-eska committed:
Translated using Weblate (Turkish)

Currently translated at 100.0% (35 of 35 strings)

Translation: bank-statement-import-18.0/bank-statement-import-18.0-account_statement_import_file
Translate-URL: https://translation.odoo-community.org/projects/bank-statement-import-18-0/bank-statement-import-18-0-account_statement_import_file/tr/

bo-eska committed:
Added translation using Weblate (Turkish)

smaddlsoft committed:
Added translation using Weblate (German)

smaddlsoft committed:
Added translation using Weblate (German)

smaddlsoft committed:
Added translation using Weblate (German)

smaddlsoft committed:
Translated using Weblate (German)

Currently translated at 100.0% (24 of 24 strings)

Translation: bank-statement-import-18.0/bank-statement-import-18.0-account_statement_import_online_ponto
Translate-URL: https://translation.odoo-community.org/projects/bank-statement-import-18-0/bank-statement-import-18-0-account_statement_import_online_ponto/de/

smaddlsoft committed:
Translated using Weblate (German)

Currently translated at 100.0% (96 of 96 strings)

Translation: bank-statement-import-18.0/bank-statement-import-18.0-account_statement_import_online
Translate-URL: https://translation.odoo-community.org/projects/bank-statement-import-18-0/bank-statement-import-18-0-account_statement_import_online/de/

smaddlsoft committed:
Translated using Weblate (German)

Currently translated at 100.0% (104 of 104 strings)

Translation: bank-statement-import-18.0/bank-statement-import-18.0-account_statement_import_sheet_file
Translate-URL: https://translation.odoo-community.org/projects/bank-statement-import-18-0/bank-statement-import-18-0-account_statement_import_sheet_file/de/

OCA-git-bot committed:
[BOT] post-merge updates

smaddlsoft committed:
Translated using Weblate (German)

Currently translated at 100.0% (104 of 104 strings)

Translation: bank-statement-import-18.0/bank-statement-import-18.0-account_statement_import_sheet_file
Translate-URL: https://translation.odoo-community.org/projects/bank-statement-import-18-0/bank-statement-import-18-0-account_statement_import_sheet_file/de/

Commit:
[UPD] Update account_statement_import_sheet_file.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: bank-statement-import-18.0/bank-statement-import-18.0-account_statement_import_sheet_file
Translate-URL: https://translation.odoo-community.org/projects/bank-statement-import-18-0/bank-statement-import-18-0-account_statement_import_sheet_file/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (105 of 105 strings)

Translation: bank-statement-import-18.0/bank-statement-import-18.0-account_statement_import_sheet_file
Translate-URL: https://translation.odoo-community.org/projects/bank-statement-import-18-0/bank-statement-import-18-0-account_statement_import_sheet_file/it/

Kiplangatdan committed:
[18.0][MIG] account_statement_import_online_wise: Migration to 18.0

- Refactor to wise.com
- Neutralize script
- Tests

bo-eska committed:
Translated using Weblate (Turkish)

Currently translated at 96.1% (101 of 105 strings)

Translation: bank-statement-import-18.0/bank-statement-import-18.0-account_statement_import_sheet_file
Translate-URL: https://translation.odoo-community.org/projects/bank-statement-import-18-0/bank-statement-import-18-0-account_statement_import_sheet_file/tr/

alexey-pelykh commented on account_statement_import_sheet_file/models/account_statement_import_sheet_parser.py:
```diff
@@ -149,17 +154,15 @@ def _get_column_names(self):
                 workbook,
-                workbook.sheet_by_index(0),
+                workbook.worksheets[0],
             )
```
`(BadZipFile, Exception)` is redundant -- `Exception` is the parent class and already covers `BadZipFile`. Just `except Exception:` works.

alexey-pelykh reviewed (commented):
Good direction -- xlrd hasn't seen active maintenance in a while and openpyxl is the standard choice now. The `_process_row_values()` extraction is a nice cleanup too.

Small thing: `except (BadZipFile, Exception)` is redundant since `Exception` already catches `BadZipFile`. Can be simplified to just `except Exception`.

Commit:
[UPD] Update account_statement_import_online_wise.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (18 of 18 strings)

Translation: bank-statement-import-18.0/bank-statement-import-18.0-account_statement_import_online_wise
Translate-URL: https://translation.odoo-community.org/projects/bank-statement-import-18-0/bank-statement-import-18-0-account_statement_import_online_wise/it/

Commit:
[UPD] Update account_statement_line_order.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (7 of 7 strings)

Translation: bank-statement-import-18.0/bank-statement-import-18.0-account_statement_line_order
Translate-URL: https://translation.odoo-community.org/projects/bank-statement-import-18-0/bank-statement-import-18-0-account_statement_line_order/it/

tarteo committed:
[FIX] account_statement_import_online_paypal: error when no transactions returned

OCA-git-bot committed:
[BOT] post-merge updates

andyengit committed:
[MERGE] various: resolve conflicts with OCA 18.0 for PR #873

Merge oca/18.0 into 18.0-openpyxl so GitHub can merge the pull request.

Contributor lists combine Anderson Armeya with upstream Jacques-Etienne
Baudoux. Parser keeps except Exception (BadZipFile redundant per review).

Made-with: Cursor

andyengit commented:
@alexey-pelykh i make the changes

lk-eska commented:
This PR become more relevant as xlrd 2.0 doesn't support xlsx....

andyengit commented:
@pedrobaeza can u check this pr?

pedrobaeza commented:
Please check instead #958

andyengit commented:
 @pedrobaeza  In this case, is it advisable to wait until the solution is available for version 19 and then port it to 18.0?

I've been using this program since last year.

pedrobaeza commented:
Sorry, I didn't see that this is for 18. Then no refactor is expected here, but the problem is that you can't change dependencies for this. Your problem is that you are using Python 3.12, while the expected Python version everywhere for 18 is 3.11. You should stick to that version for not having problems IMO. If not, you should do an extra module for using the XLSX with openpyxl.

lk-eska commented:
why don't we just accept this for 3.12 users

andyengit commented:
@pedrobaeza  I'll keep that in mind. I'm new to working with OCA. Do you recommend taking the extraction request currently being made for version 19.0 and migrating it to 18.0? I see you're creating a module specifically for xls and xlsx.

Or would there be no problem with how I handle this solution in 18.0?

pedrobaeza commented:
Because it will break all the existing installations (not deployed with PyPi), changing the dependencies of an stable module.

alexey-pelykh reviewed (commented):
Thanks @andyengit — re-reviewed against the current diff.

My earlier nit is resolved: the exception handling is now just `except Exception:`, no more redundant `(BadZipFile, Exception)`. 👍

The underlying need is real — with xlrd 2.0 no longer reading `.xlsx` at all, `openpyxl` is the right engine going forward. But as a maintainer of `account_statement_import_sheet_file`, I share @pedrobaeza's concern about the approach: swapping the dependency (`xlrd` → `openpyxl`) inside this *stable* 18.0 module changes its runtime deps and breaks existing installations that don't resolve deps from PyPI. On a stable branch we can't do that in place.

The non-breaking path for 18.0 is a separate module that layers openpyxl-based parsing on top of this one — the same pattern #871 uses to add HTML support — rather than replacing the parser here. (The 19.0 refactor already merged in #958.)

Two practical blockers on the current state as well:

- `codecov/patch` and `codecov/project` are red — the new `_process_row_values()` and the openpyxl branch in `_parse_rows` aren't covered (no tests were added in the PR).
- The branch is conflicting with `18.0` and needs a rebase.

The openpyxl parsing itself looks reasonable — it's really the packaging for a stable branch that needs to change. Happy to review a standalone-module version.

Co-Reviewed-By: Claude Opus 4.8 <noreply@anthropic.com>
